### PR TITLE
Third problem

### PR DESCRIPTION
**Pull request by HiciuStefan**

### CODE REVIEW

File: third_problem.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class ThirdProblem
+  def initialize(numbers_array)
+    @numbers = numbers_array
+  end
+
+  def hasDuplicates
```
Comment (alexandru-calinoiu) [inline third_problem.rb:6]: Much better :+1:

File: third_problem.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    found = false
+    @numbers.each {|index|
+      absolute_current_index =index.abs-1
+      absolut_target_index = @numbers[absolute_current_index].abs-1
```
Comment (alexandru-calinoiu) [inline third_problem.rb:10]: You keep repeating `@numbers[absolute_current_index]`, I will think is worth extracting in a variable. Index lookup for arrays in ruby has it's cost.

File: third_problem.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      if @numbers[absolut_target_index] < 0
+        found = true
+        break
+      end
+      @numbers[absolut_target_index] = -@numbers[absolut_target_index]
+    }
+    found
```
Comment (alexandru-calinoiu) [inline third_problem.rb]: If you where allowed to use a different data structure to solve this problem what data structure will you use and why?

File: third_problem.rb (hunk 2 above)
Comment (HiciuStefan) [reply third_problem.rb:10]: sure , extracted to variable

**Comment by alexandru-calinoiu** [comment]
Thank you, will write a email shortly.

**Pull request closed without merging**